llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -541,6 +541,33 @@ void llvm::processShuffleMasks(
   }
 }
 
+void llvm::getHorizDemandedEltsForFirstOperand(unsigned VectorBitWidth,
+                                               const APInt &DemandedElts,
+                                               APInt &DemandedLHS,
+                                               APInt &DemandedRHS) {
+  int NumLanes = std::max<int>(1, VectorBitWidth / 128);
```

Comment:
Can you just assert VectorBitWidth >= 128? We where assuming that before.

Reply:
Otherwise add a few cases w/ width < 128 in your unit tests.

Reply:
Looks like we're still missing MMX test coverage - I have no preference to whether you add them to the valuetracking tests or drop the sub-128 handling.

Reply:
It seems that KnownBits computation is very limited for `x86_mmx`. Hence, I think, it's not worth the trouble to keep the computation for the MMX variants of `phadd`. Therefore, I've removed the support and added the suggested assert.